tests/unit/plugins/modules/test_idrac_support_assist.py

```diff
@@ -0,0 +1,1377 @@
+        eula_status_support_assist_obj = self.module.AcceptEULA(
+            idrac_connection_support_assist_mock, f_module)
+        eula_status_support_assist_obj._AcceptEULA__get_eula_status_url()
+        assert eula_status_support_assist_obj.eula_status_url == API_ONE
+
+        # Scenario 2: When url is empty for Links
+        mocker.patch(MODULE_PATH + "get_dynamic_uri",
+                     return_value={"Links": {}})
+        with pytest.raises(Exception) as exc:
+            eula_status_support_assist_obj._AcceptEULA__get_eula_status_url()
+        assert exc.value.args[0] == UNSUPPORTED_FIRMWARE_MSG
+
+        # Scenario 3: For error message
+        mocker.patch(MODULE_PATH + "validate_and_get_first_resource_id_uri",
+                     return_value=(REDFISH, "error"))
+        with pytest.raises(Exception) as exc:
+            eula_status_support_assist_obj._AcceptEULA__get_eula_status_url()
+        assert exc.value.args[0] == "error"
+
+    def test_get_eula_accept_url(self, idrac_default_args, idrac_connection_support_assist_mock, mocker):
+        mocker.patch(MODULE_PATH + "validate_and_get_first_resource_id_uri",
+                     return_value=(REDFISH, None))
+        # Scenario 1: With url
+        mocker.patch(MODULE_PATH + "get_dynamic_uri",
+                     return_value={"Links": {"Oem": {"Dell": {"DellLCService": {ODATA: ASSIST_ODATA}}}},
+                                   "Actions": {"#DellLCService.SupportAssistAcceptEULA": {"target": API_ONE}}})
+        f_module = self.get_module_mock(
+            params=idrac_default_args, check_mode=False)
+        eula_accept_obj = self.module.AcceptEULA(
+            idrac_connection_support_assist_mock, f_module)
+        eula_accept_obj._AcceptEULA__get_eula_accept_url()
+        assert eula_accept_obj.eula_accept_url == API_ONE
+
+        # Scenario 2: When url is empty for Links
+        mocker.patch(MODULE_PATH + "get_dynamic_uri",
+                     return_value={"Links": {}})
+        with pytest.raises(Exception) as exc:
+            eula_accept_obj._AcceptEULA__get_eula_accept_url()
+        assert exc.value.args[0] == UNSUPPORTED_FIRMWARE_MSG
+
+        # Scenario 3: For error message
+        mocker.patch(MODULE_PATH + "validate_and_get_first_resource_id_uri",
+                     return_value=(REDFISH, "error"))
+        with pytest.raises(Exception) as exc:
+            eula_accept_obj._AcceptEULA__get_eula_accept_url()
+        assert exc.value.args[0] == "error"
+
+    def test_eula_status(self, idrac_default_args, idrac_connection_support_assist_mock, mocker):
+        obj = MagicMock()
+        mocker.patch(MODULE_PATH + API_INVOKE_MOCKER, return_value=obj)
+        f_module = self.get_module_mock(
+            params=idrac_default_args, check_mode=False)
+        eula_status_obj = self.module.AcceptEULA(
+            idrac_connection_support_assist_mock, f_module)
+        status = eula_status_obj.eula_status()
+        assert status
```

Comment:
assert like 

status is True or False

Reply:
it returns magicmock id, can't assert that.